validator/test/fixtures/sql_maintenance_window/assets/data.json

```diff
@@ -0,0 +1,110 @@
+               "backupConfiguration":{  
+                  "enabled":true,
+                  "kind":"sql#backupConfiguration",
+                  "replicationLogArchivingEnabled":false,
+                  "startTime":"04:00"
+               },
+               "dataDiskSizeGb":"10",
+               "dataDiskType":"PD_SSD",
+               "ipConfiguration":{  
+                  "ipv4Enabled":true,
+                  "privateNetwork":"projects/test-project/global/networks/default"
+               },
+               "kind":"sql#settings",
+               "locationPreference":{  
+                  "kind":"sql#locationPreference",
+                  "zone":"us-central1-a"
+               },
+               "maintenanceWindow":{  
+                  "day":1,
+                  "hour":0,
+                  "kind":"sql#maintenanceWindow"
+               },
+               "pricingPlan":"PER_USE",
+               "replicationType":"SYNCHRONOUS",
+               "settingsVersion":"0",
+               "storageAutoResize":true,
+               "storageAutoResizeLimit":"0",
+               "tier":"db-custom-1-3840"
+            },
+            "state":"PENDING_CREATE"
+         }
+      }
+   },
+   {  
+      "name":"//cloudsql.googleapis.com/projects/test-project/instances/maintenance-sql-b",
+      "asset_type":"sqladmin.googleapis.com/Instance",
+      "resource":{  
+         "version":"v1beta4",
+         "discovery_document_uri":"https://www.googleapis.com/discovery/v1/apis/sqladmin/v1beta4/rest",
+         "discovery_name":"DatabaseInstance",
+         "data":{  
+            "backendType":"SECOND_GEN",
+            "connectionName":"test-project:us-central1:maintenance-sql-b",
+            "databaseVersion":"POSTGRES_9_6",
+            "gceZone":"us-central1-a",
+            "instanceType":"CLOUD_SQL_INSTANCE",
+            "kind":"sql#instance",
+            "name":"maintenance-sql-b",
+            "project":"test-project",
+            "region":"us-central1",
+            "selfLink":"https://www.googleapis.com/sql/v1beta4/projects/test-project/instances/maintenance-sql-b",
+            "settings":{  
+               "activationPolicy":"ALWAYS",
+               "availabilityType":"ZONAL",
+               "backupConfiguration":{  
+                  "enabled":true,
+                  "kind":"sql#backupConfiguration",
+                  "replicationLogArchivingEnabled":false,
+                  "startTime":"22:00"
+               },
+               "dataDiskSizeGb":"10",
+               "dataDiskType":"PD_SSD",
+               "ipConfiguration":{  
+                  "ipv4Enabled":false,
+                  "privateNetwork":"projects/test-project/global/networks/default"
+               },
+               "kind":"sql#settings",
+               "locationPreference":{  
+                  "kind":"sql#locationPreference",
+                  "zone":"us-central1-a"
+               },
+               "maintenanceWindow":{  
+                  "day":0,
```

Comment:
Is this real data? From what I can tell, when you don't set a maintenance window for you SQL instance, the maintenanceWindow key is abstent from the data.

Reply:
Now with real data. SQL first generation do not support maintenance window and are filtered out. for 2nd generation the maintenanceWindow key always exit. day is set to 0 when no maintenance window defined.